File: MinimumTimetoMakeRopeColorful/MinimumTimetoMakeRopeColorful.java

import java.util.LinkedList;
import java.util.List;

public class MinimumTimetoMakeRopeColorful {

    private class Solution{
    public int minCost(String colors, int[] neededTime) {
        int max_value=0,n=colors.length();
        List<List<Integer>> stack=new LinkedList<>();
        List<Integer> list= new LinkedList<>();
        list.add((int)colors.charAt(0));
        list.add(neededTime[0]);
        stack.add(list);
        for(int i=1;i<n;i++){
            List<Integer> new_item=new LinkedList<>();
            new_item.add((int)colors.charAt(i));
            new_item.add(neededTime[i]);    
            List<Integer> last_item=stack.get(stack.size()-1);
            if(last_item.get(0)==(int)colors.charAt(i)){
                 if(last_item.get(1)<neededTime[i]){
                    max_value+=last_item.get(1);
                    stack.remove(stack.size()-1);
                    stack.add(new_item);
                 }else{
                    max_value+=neededTime[i];
                 }

            }else{
                stack.add(new_item);
            }
        }
        return max_value;
    }
    }
}

File: MinimumTimetoMakeRopeColorful/MinimumTimetoMakeRopeColorful.kt
class Solution {
    fun minCost(colors: String, neededTime: IntArray): Int {
        val n: Int = colors.length
        var max_value: Int = 0
        var stack: MutableList<MutableList<Int>> =mutableListOf(mutableListOf(colors.get(0).code, neededTime[0]));
        for(i in 1..<n){
            val new_item=mutableListOf(colors.get(i).code,neededTime.get(i));
            val old_item=stack.get(stack.size-1);
            if(old_item.get(0)==colors.get(i).code){
                
               if(old_item.get(1)<neededTime.get(i)) {

                   max_value+=old_item.get(1);
                   stack.removeAt(stack.size-1);
                   stack.add(new_item);
                }
                else{
                    max_value+=neededTime.get(i);
    
                }

            }else{
                stack.add(new_item);
            }
        }

        return max_value;   

    }
}
